projects/packages/publicize/src/class-rest-controller.php

```diff
@@ -680,6 +680,28 @@ public function get_post_share_status( WP_REST_Request $request ) {
 		// If the data is not an array, it means that sharing is not done yet.
 		$done = is_array( $shares );
 
+		if ( $done ) {
+			// The site could have multiple admins, editors and authors connected. Load shares information that only the current user has access to.
+			global $publicize;
+			$connection_ids = array_map(
+				function ( $connection ) {
+					if ( isset( $connection['connection_id'] ) ) {
+						return (int) $connection['connection_id'];
+					}
+					return 0;
+				},
+				$publicize->get_all_connections_for_user()
+			);
+			$shares         = array_values(
+				array_filter(
+					$shares,
+					function ( $share ) use ( $connection_ids ) {
+						return in_array( (int) $share['connection_id'], $connection_ids, true );
+					}
+				)
+			);
+		}
```

Comment:
It means that if a connection is reconnected, the previous shares for that connection won't be shown, is that what we want?

Reply:
Should we use the same logic here that we use on the frontend via [`connectionMatchesShareItem`](https://github.com/Automattic/jetpack/blob/57aef230c221b772e4294a5f1ada2e447179645f/projects/js-packages/publicize-components/src/utils/share-status.ts#L18-L40)?

Reply:
>Should we use the same logic here that we use on the frontend via [connectionMatchesShareItem](https://github.com/Automattic/jetpack/blob/57aef230c221b772e4294a5f1ada2e447179645f/projects/js-packages/publicize-components/src/utils/share-status.ts#L18-L40)?

I thought of that, but then decided to go against it and keep it simple because the following snippet could still return another user's share log. 2 Admins can still connect the same social  account technically, if not at the same time, the second admin can connect it after the first admin removes it. They still should not see each other's logs if the connection was not shared. 


```
	// external_id may not be present in old data, so we need to check for it
		if ( shareItem.external_id ) {
			return connection.external_id === shareItem.external_id;
		}
```

Reply:
Fair enough